cinema/views.py

```diff
@@ -24,21 +27,53 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound("Method not allowed for this resource")
```

Comment:
Raising `NotFound` for unsupported actions is misleading. The `MethodNotAllowed` exception should be used instead to indicate that the HTTP method is not allowed for the requested resource.